lib/helpers.rb

```diff
@@ -232,7 +239,7 @@ def handle_threads_query(
         pres_threads = ThreadListPresenter.new(threads, request_user, course_id)
         collection = pres_threads.to_hash
       end
-      {collection: collection, num_pages: num_pages, page: page, thread_count: comment_threads.count}
+      {collection: collection, num_pages: num_pages, page: page, thread_count: thread_count}
```

Comment:
I am not an expert in ruby, but I am asked to take a look at this pr. 

Just a question, before this change, `comment_threads.count` is supposed to send `thread_count` for all comment_threads and now, `thread_count` will only have the count of paginated results. 

Is this expected?

Reply:
paginated_collection.total_entries stores total count of threads passed to it for pagination. Total count is required to calculate per page threads. So, we have used library's variable 'total_entries' as it does the same job; it fetches the total count of threads. We have used that library's variable 'total_entries' to avoid performing count operation explicity, as it was done before e.g. 'comment_threads.count' because it is a costly operation when the collection/threads size is big.

Reply:
Sure, that sounds good. Thank you for the examination.